test/distributed/test_composability.py

```diff
@@ -204,21 +237,21 @@ def build_stage(stage_idx, num_stages):
                 loss_fn=loss_fn,
             )
 
-        # Run
-        # TODO(whc) should we make it a hard error if you pass arguments into the step API on nonzero ranks?
-        # why are we passing inputs/targets on every rank?
+        # Run the pipeline
         if pp_group.rank() == 0:
-            pipeline_schedule._step_microbatches(arg_mbs=input_mb, target_mbs=input_mb)
+            pipeline_schedule._step_microbatches(
+                arg_mbs=input_mb, target_mbs=[[] for _ in target_mb]
```

Comment:
nit: for rank 0, does target_mbs need to get passed in here? Thought that it was skipped

Reply:
oops missed this. I think you're right, I should omit the argument entirely.  I have another PR for more refactors anyways, i'll address this in that PR.